website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+            }
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig});
+            promise.then(success, error);
+            return promise;
+        };
+        self.updateListSuccess = function _updateListSuccess (value) {
+            if(self.loadingNodePages){
+                self.data().data = self.data().data.concat(value.data);
+            } else {
+                self.data(value);
+            }
+            if(!value.data[0]){ // If we have projects
+                var lastcrumb = self.breadcrumbs()[self.breadcrumbs().length-1];
+                if(lastcrumb.type === 'collection'){
+                    if(lastcrumb.data.systemCollection === 'nodes'){
+                        self.nonLoadTemplate(m('.fb-non-load-template.m-md.p-md.osf-box',
+                            'You have not created any projects yet.'));
+                    } else if (lastcrumb.data.systemCollection === 'registrations'){
+                        self.nonLoadTemplate(m('.fb-non-load-template.m-md.p-md.osf-box',
+                            'You have not made any registrations yet.'));
+                    } else {
+                        self.nonLoadTemplate(m('.fb-non-load-template.m-md.p-md.osf-box',
+                            'This collection has no projects. To add projects go to "All My Projects" collection; drag and drop projects into the collection link'));
+                    }
+                } else {
+                    self.nonLoadTemplate(m('.fb-non-load-template.m-md.p-md.osf-box.text-center', [
+                        'This project has no subcomponents.',
+                        m.component(AddProject, {
+                            buttonTemplate : m('.btn.btn-link[data-toggle="modal"][data-target="#addSubcomponent"]', 'Add new Subcomponent'),
+                            parentID : self.breadcrumbs()[self.breadcrumbs().length-1].data.id,
+                            modalID : 'addSubcomponent',
+                            stayCallback : function _stayCallback_inPanel() {
+                                self.allProjectsLoaded(false);
+                                self.updateList(self.breadcrumbs()[self.breadcrumbs().length-1]);
+                            }
+                        })
+                    ]));
+                }
+            }
+            // if we have more pages keep loading the pages
+            if (value.links.next) {
+                self.loadingNodePages = true;
+                var collData = {};
+                if(!self.allProjectsLoaded()) {
+                    collData = { systemCollection : 'nodes' };
+                }
+                self.updateList({link : value.links.next, data : collData });
+                return; // stop here so the reloads below don't run
+            } else {
+                self.loadingNodePages = false;
+            }
+            if(self.loadingAllNodes) {
+                self.data().data = _makeTree(self.data().data);
```

Comment:
I'm not sure these direct modifications of `self.data().data` will behave as expected. Mithril props are meant to be called when assigning, so that controllers will properly re-instantiate. As suggested in https://github.com/CenterForOpenScience/osf.io/pull/4887/files#r51177967, better to have `self.data()` be the unwrapped data, so that you can do

```js
self.data(_makeTree(self.data()));
```

which will behave more predictably.